github/data_source_github_repository_test.go

```diff
@@ -55,29 +55,35 @@ func TestAccGithubRepositoryDataSource(t *testing.T) {
 
 		config := map[string]string{
 			"no_match_name": `
-				data "github_repository" "no_match_name" {
+				data "github_repository" "test" {
 					name = "owner/repo"
 				}
 			`,
 			"no_match_fullname": `
-				data "github_repository" "no_match_fullname" {
+				data "github_repository" "test" {
 					full_name = "owner/repo"
 				}
 			`,
 		}
 
+		check := resource.ComposeTestCheckFunc(
+			resource.TestCheckNoResourceAttr(
+				"data.github_branch.test", "id",
```

Comment:
I'm not actually sure what this is checking. Do you know?

Reply:
I think I was under the impression that it made sure that the repository doesn't exist (e.g. has no ID). Probably, my idea was that if that holds true and the test doesn't crash (from the missing actual repository) then it proves that the feature works. But, yeah, this is far from clear and definitely needs reworking.